### ISSUE TITLE
NVDA doesn't announce keystrokes in Windows Calculator

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#3841]
**Reported by supanut2000 on 2014-02-03 10:09**
In the Windows Calculator in Windows 7, NVDA doesn't announce the keystrokes, for example when I press the shortcut key for sine, cosine, tangin, etc.

**Comment by leonardder** [comment]
Hmm, is NVDA supposed to announce these shortcut keys? NVDA should only do this when speak command keys is on, and the reporter didn't provide this information unfortunately.

**Comment by Adriani90** [comment]
@supanut2000 can you reproduce this issue still, even with enabled "speak command keys" in NVDA keyboard settings?

**Comment by Adriani90** [comment]
Testing with NVDA 2019.1.1 on Windows 10 calculator it works as follows:
1. Open calculator
2. Switch to scientific calculator with alt+2
3. Press ctrl+o for cos, ctrl+t for tan or ctrl+s for sin or shift+s for inverse sin.

Note that you must press every of the key strokes above twice in order to add the expression. It is properly reported by NVDA.

@k-kolev1985 could you please test on Windows 8.1 in the calculator if it works?

For additional shortcuts see this treat:
https://superuser.com/questions/546232/how-do-i-view-windows-8-calculator-shortcut-keys

**Comment by k-kolev1985** [comment]
When I Press keystrokes like Control+O, Control+S, etc. in the scientific view of the windows 8.1 calculator, NVDA reports only the digital result. It does not report the text above the result in the calculator's display.

**Comment by Adriani90** [comment]
Hmm strange. For me on Windows 10 NVDA reports what I have added to the function (i.e. cosinus hyperbolic function). Pressing enter after that NVDA reports the result of the function.
For example:
1. Open the calculator
2. Press alt+2
3. Type 2+4 and press enter (you will hear the result which is = 6)
4. Press ctrl+o twice to put 6 in a double cosinus function (you will hear "the display shows hyperbolic cosinus, hyperbolic cosinus (6))
5. Press enter (you will hear result 2.008... etc..) which is the result of the cosinus(cosinus(6)) function.

Could you please post here what NVDA reports if you do the same steps like me above? I don't know exactly what hapens there but it seems NVDA does not report the first hyperbolic cosinus. It begins to report when pressing ctrl+o two or three times.
cc: @josephsl

**Comment by k-kolev1985** [comment]
1. Which calculator in Windows 8.1? The last time I tested with the "modern" one. Is that the one, or you mean the desktop/classic one? Now I've tested with both.
2. In the modern calculator, Alt+2 seams not to work - I had to activate the "Scientific" mode manually by activating its tab.
3. Here NVDA reports the calculation result as it should.
4. When pressing Control+O twice, the calculators report invalid input, but NVDA does not read/echo this, even though it is shown on their displays.
5. If I remember correctly, pressing ENTER does not report anything.

The behavior described by you occurs in the Windows 10 calculator, but seams not to in the ones in Windows 8.1.

**Comment by josephsl** [comment]
Hi,

2022 update: now I know why: same underlying issue as #9428 - in Windows 10/11 Calculator, across calculator modes, the same activity Id is used to display results. Fixing this will require changing UIA notification event handler to silence "display is ..." announcement when you press just single keys (numbers, punctuation, etc.). Note that this assumes Windows 10 - if it is Windows 8.1, then it may require a different approach as old Calculator is not really the same app as modern Calculator.

Thanks.

**Comment by Adriani90** [comment]
@josephsl I think it makes sense to concentrate on Windows 10 and 11 now, Windows 8.1 is not really relevant anymore.

Do you have any update in 2024 with NVDA 2024.3 Beta?

**Comment by josephsl** [comment]
Hi,

It would be best to ask others who have investigated this issue as well, not just yours truly.

Thanks.

**Comment by Adriani90** [comment]
This seems not reproducible on my side on windows 10 and 11 with last NVDA alpha. I am closing as works for me. if you are still having this issue, please comment and we can reopen.
